src/main.dart

```diff
@@ -1,8 +1,142 @@
 // TODO: Run task list app
-void main(List<String> args) {
-  print('Hello World');
+import 'taskClass.dart';
+import 'dart:io';
+
+void printMenu() {
+  print('\n-------------------------');
+  print('-     Task List App     -');
+  print('-------------------------');
+  print('- 1. Add a Task         -');
+  print('- 2. Edit a Task        -');
+  print('- 3. Delete a Task      -');
+  print('- 4. Mark as Completed  -');
+  print('- 5. Print Tasks List   -');
+  print('- 6. Exit               -');
+  print('-------------------------\n');
+}
+
+void addTask() {
+  stdout.write('\nEnter Task title: ');
+  String? t = stdin.readLineSync();
+  stdout.write('\nEnter Task Description: ');
+  String? d = stdin.readLineSync();
+  TaskListApp.tasks.add(new TaskListApp(t!, d!));
+}
+
+void editTask(int no) {
+  if (TaskListApp.tasks.elementAtOrNull(no) != null) {
+    stdout.write('\nEdit Task title: ');
+    String? t = stdin.readLineSync();
+    stdout.write('\nEdit Task Description: ');
+    String? d = stdin.readLineSync();
+    TaskListApp.tasks.elementAt(no).title = t!;
+    TaskListApp.tasks.elementAt(no).description = d!;
+  } else {
+    print('Task dose not exist!\n');
+  }
+}
+
+void deleteTask(int no) {
+  TaskListApp.tasks.remove(no);
+  TaskListApp.count--;
+  for (int i = no; i < TaskListApp.tasks.length; i++) {
+    TaskListApp.tasks.elementAt(i + 1).id = i;
+  }
+}
+
+void markCompleted(int no) {
+  if (TaskListApp.tasks.elementAtOrNull(no) != null) {
+    TaskListApp.tasks.elementAt(no).isCompleted = true;
+    print('${TaskListApp.tasks.elementAt(no).title} task is Completed! ;) ');
+  } else {
+    print('Task dose not exist!\n');
+  }
 }
 
-class TaskListApp {
-  // TODO: Implement Task list app
+void printTasksList() {
+  for (int i = 0; i < TaskListApp.tasks.length; i++) {
+    print(
+        '${TaskListApp.tasks.elementAt(i).id}. ${TaskListApp.tasks.elementAt(i).title}');
+    print('Description: ${TaskListApp.tasks.elementAt(i).description}');
+    if (TaskListApp.tasks.elementAt(i).isCompleted) {
+      print('Completion Status: Completed\n');
+    } else {
+      print('Completion Status: In Progress\n');
+    }
```

Comment:
use one line if instead
```suggestion
    print("Completion Status: ${TaskListApp.tasks.elementat(i).isCompleted ? "Completed" : "In Progress"})
```